app/ammunition/potion.py

```diff
@@ -0,0 +1,25 @@
+class Potion:
+    """"
+    Knight's potion.
+
+    Something can be not mentioned, so checked for None.
+    """
+    def __init__(self, name: str, effect: dict) -> None:
+        self.name = name
+        self.hp_boost = effect.get("hp")
+        self.power_boost = effect.get("power")
+        self.protection_boost = effect.get("protection")
+        if self.hp_boost is None:
```

Comment:
You can simply assign variable inline, because get() and pass default parameter to get("hp", 0) for example

Reply:
Removed none-check. get default variable 0 is setted